base/iostream.jl

```diff
@@ -225,8 +225,18 @@ function readuntil(s::IOStream, delim::UInt8)
     ccall(:jl_readuntil, Array{UInt8,1}, (Ptr{Void}, UInt8, UInt8), s.ios, delim, 0)
 end
 
-function readline(s::IOStream)
-    ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+function readline(s::IOStream, chomp::Bool=false)
+    chomp || return ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+
+    s = ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+    i = endof(s)
+    if i < 1 || codeunit(s,i) != 0x0a
+        return s[1:i]
```

Comment:
Too bad. We may as well reintroduce `jl_readline` from https://github.com/JuliaLang/julia/pull/19877 for now, since @mpastell has already written it. It will be easier to remove when possible than modifying `jl_readuntil`.